opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/_load.py

```diff
@@ -0,0 +1,124 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from logging import getLogger
+from os import environ
+
+from pkg_resources import iter_entry_points
+
+from opentelemetry.instrumentation.dependencies import (
+    get_dist_dependency_conflicts,
+)
+from opentelemetry.instrumentation.distro import BaseDistro, DefaultDistro
+from opentelemetry.instrumentation.environment_variables import (
+    OTEL_PYTHON_CONFIGURATOR,
+    OTEL_PYTHON_DISABLED_INSTRUMENTATIONS,
+    OTEL_PYTHON_DISTRO,
+)
+from opentelemetry.instrumentation.version import __version__
+
+_logger = getLogger(__name__)
+
+
+def _load_distros() -> BaseDistro:
+    distro_name = environ.get(OTEL_PYTHON_DISTRO, None)
```

Comment:
Since we're only one level away from the bottom of the call stack here, what do you think about making `distro_name` an argument to this function and having the caller pass in the environment variable value?

Also, while we're in here, should the name of the function be `_load_distro()` since it only appears to load one?

Reply:
Good point. Will change

Reply:
Done.

Reply:
Missed the first part of this. I changed the name to _load_distro(). @pmcollins Could you explain the purpose of changing function signature? I think since we cannot test sitecustomize itself, it is nice to have all the functionality in the function itself.

Reply:
Ok, no worries.

My vision though, is to nudge the code in the direction of reading environment variables in one place at the beginning of a program's lifecycle (in main or the equivalent, which in this case is sitecustomize.py) and then passing those values as function arguments, rather than reading environment variables at specific points in the call stack only when they are needed.

To this end, environment variables could be read directly in `initialize()` and then passed as arguments to the functions that need those values. However, in the longer term, it would be nice to have a separate, independently testable loader for environment variables. This would allow us to modify the environment in our tests without affecting the global state, which could be beneficial for testing parts of the code that depend on these variables.

In any case, we can address this separately if folks are interested. I am open to other perspectives on this, and I am happy to discuss more if needed.

Reply:
👆 Not a blocking comment btw. Discussed this with Jeremy and environment variable management is something that we can consider separately.